### ISSUE TITLE
Trouble with crazyflie2 hovering example. [Ubuntu 18.04, ROS Melodic, Gazebo 9] 

### ISSUE THREAD

**Issue by umidah** [gsilano/CrazyS#90]
Good day, I installed CrazyS for ubuntu 18.04, ROS melodic and Gazebo9. When running the crazyflie2 hovering example, I could not get it to run.

I used the example command given in the readme:
```bash
roslaunch rotors_gazebo crazyflie2_hovering_example.launch
```

This was the error displayed for me.
```bash
... logging to /home/ajtpa/.ros/log/cdbdd2ee-835f-11ec-989a-e0d55ea1eb65/roslaunch-otonokizaka-11207.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

xacro: in-order processing became default in ROS Melodic. You can drop the option.
name 'namespace' is not defined 
when evaluating expression 'namespace'
when processing file: /home/ajtpa/catkin_ws/src/CrazyS/rotors_description/urdf/component_snippets.xacro
included from: /home/ajtpa/catkin_ws/src/CrazyS/rotors_description/urdf/crazyflie2_base.xacro
included from: /home/ajtpa/catkin_ws/src/CrazyS/rotors_description/urdf/mav_generic_odometry_sensor.gazebo
RLException: while processing /home/ajtpa/catkin_ws/src/CrazyS/rotors_gazebo/launch/spawn_mav_crazyflie.launch:
Invalid <param> tag: Cannot load command parameter [robot_description]: command [['/opt/ros/melodic/lib/xacro/xacro', '--inorder', '/home/ajtpa/catkin_ws/src/CrazyS/rotors_description/urdf/mav_generic_odometry_sensor.gazebo', 'enable_logging:=false', 
```

If any more information is needed, I would gladly provide it.

EDIT: This was done in a fresh install of Ubuntu 18.04

**Comment by rglbiv** [comment]
Hello, I got the same issue that returns the same error message. I followed the instruction in the README.md for Ubuntu 18.04.

**Comment by gsilano** [comment]
It seems that the problem lies in some hold definitions contained in the `spawn_mav_crazyflie.launch`. Open the file, and add the `--inorder` option as described in here https://answers.ros.org/question/354092/xacro-in-order-processing-for-husky_gazebo/. I hope this helps.

**Comment by umidah** [comment]
Thanks for the reply, I'll check it out. If it does not work, I might try re-installing with ubuntu 20.04 and ROS noetic.

**Comment by rglbiv** [comment]
Thanks for the quick reply. Unfortunately, the --inorder argument is already in the file as seen below.
![image](https://user-images.githubusercontent.com/46555394/152010233-e34e4208-0fdd-4096-980f-6b637b942477.png)

If I try to launch the file with or without the --inorder, the same error appears.
![image](https://user-images.githubusercontent.com/46555394/152009618-a8f9baa7-2765-47ce-b24a-4bcb76d0e77e.png)

**Comment by gsilano** [comment]
Looking at your screens, especially the second, it seems there is a problem with the namespace. Could you try to run any other launch file in the repository? I would like to understand if this problem affects only the file you mentioned or all the scripts have suffer of the same issue. Thanks!

**Comment by rglbiv** [comment]
I tried it with:
mav_with_keyboard.launch and crazyflie2_swarm_hovering_example.launch, which also suffers from the issue.
![image](https://user-images.githubusercontent.com/46555394/152014122-87903eb2-cdfc-4e85-bfd7-0a8e44068848.png)

**Comment by gsilano** [comment]
Ok, so everything seems to be right with the ROS environment. Have you tried to run the Crazyflie example without the controller?

**Comment by rglbiv** [comment]
If I try to run the file and comment the controller part, this is the result:
![image](https://user-images.githubusercontent.com/46555394/152082002-ec70e2d5-55ab-45bb-a4e7-7c35072d7d4a.png)

**Comment by gsilano** [comment]
Ok! Let me test it on my PC. I wil give you back in a short time.

**Comment by rglbiv** [comment]
Good day sir, are there updates to a fix to this issue? Thanks!

**Comment by rglbiv** [comment]
I tried reinstalling it again in a fresh machine, and the error is still there. Problem regarding namespace.
![image](https://user-images.githubusercontent.com/46555394/152969443-28a807ff-ef30-40a1-934e-1a9afab8c40c.png)

**Comment by rglbiv** [comment]
I tried installing it with Ubuntu 20.04 and it works with no issues, I guess there's a problem within Ubuntu 18.04. Thanks for your time.

**Comment by gsilano** [comment]
Ok great! Did you compare the launch files? I'm curious if the problem lies there, or is it just a problem with ROS. However, this incompatibility problem is new to me.

**Comment by rglbiv** [comment]
Unfortunately no as I have already deleted my Ubuntu 18.04 to install the new version.

**Comment by gsilano** [comment]
Ah, okay. Finally you managed to make it works, and this what really matters ;-) Enjoy!

**Comment by anavsalkar** [comment]
Hello, I am facing the same issue on my 20.04 system. I correctly installed as per the instructions for Ubuntu 20.04.
![image](https://user-images.githubusercontent.com/47375604/153725879-3c4277cd-985a-418b-9422-550039036fc3.png)
I tried changing the "<param name="robot_description"......" in the "spawn_mav.launch" file including/ommiting '--inorder' or 'xacro' or 'xacro.py' as suggested in ( https://answers.ros.org/question/122021/xacro-problem-invalid-param-tag-cannot-load-command-parameter-robot_description/ ), without any success. 
What's surprising is that I had no trouble working with RotorS installation from 'eth-asl' when I last used it 2 months back. Now, both the installations (earlier RotorS from eth-asl and CrazyS) give the same errors. Don't know if some update has caused this.
Thank you in advance for any help!

**Comment by anavsalkar** [comment]
I managed to solve the issue. Well, like I said it was indeed due to version updates. I removed `xacro` and `xacro-1.14.11.egg-info` directories from `/opt/ros/noetic/lib/python3/dist-packages` folder using `sudo rm -r ...` command and installed the 1.13.3 version of xacro using `pip3 install xacro==1.13.3`. This simple downgrading from 1.14.11 to 1.13.3 worked for me, without any changes in launch files.
